src/ephemeral.rs

```diff
@@ -212,21 +212,89 @@ pub(crate) async fn stock_ephemeral_timer_changed(
     timer: Timer,
     from_id: u32,
 ) -> String {
-    let stock_message = match timer {
-        Timer::Disabled => StockMessage::MsgEphemeralTimerDisabled,
+    match timer {
+        Timer::Disabled => {
+            context
+                .stock_system_msg(
+                    StockMessage::MsgEphemeralTimerDisabled,
+                    timer.to_string(),
+                    "",
+                    from_id,
+                )
+                .await
+        }
         Timer::Enabled { duration } => match duration {
-            60 => StockMessage::MsgEphemeralTimerMinute,
-            3600 => StockMessage::MsgEphemeralTimerHour,
-            86400 => StockMessage::MsgEphemeralTimerDay,
-            604_800 => StockMessage::MsgEphemeralTimerWeek,
-            2_419_200 => StockMessage::MsgEphemeralTimerFourWeeks,
-            _ => StockMessage::MsgEphemeralTimerEnabled,
+            0..=59 => {
+                context
+                    .stock_system_msg(
+                        StockMessage::MsgEphemeralTimerEnabled,
+                        timer.to_string(),
+                        "",
+                        from_id,
+                    )
+                    .await
+            }
+            60 => {
+                context
+                    .stock_system_msg(StockMessage::MsgEphemeralTimerMinute, "", "", from_id)
+                    .await
+            }
+            61..=3599 => {
+                context
+                    .stock_system_msg(
+                        StockMessage::MsgEphemeralTimerMinutes,
+                        format!("{}", (f64::from(duration) / 6.0).round() / 10.0),
```

Comment:
that's a clever way of doing formatting with one optional digit after the comma :)